src/extension/app/components/onboarding/onboarding-dialog.js

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { html } from '@spectrum-web-components/base';
+import { queryAsync, state } from 'lit/decorators.js';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import { when } from 'lit/directives/when.js';
+import { style } from './onboarding-dialog.css.js';
+import { ConnectedElement } from '../connected-element/connected-element.js';
+import { ICONS } from '../../constants.js';
+import { getConfig, setConfig } from '../../../config.js';
+
+/**
+ * The onboarding item type
+ * @typedef {import('@Types').OnboardingItem} OnboardingItem
+ */
+
+/**
+ * Spectrum tabs
+ * @typedef {import('@spectrum-web-components/tabs').Tabs} Tabs
+ */
+
+/**
+ * Spectrum dialog
+ * @typedef {import('@spectrum-web-components/dialog').DialogBase} DialogBase
+ */
+
+const TOOLS_ORIGIN = 'https://tools.aem.live';
+
+export class OnBoardingDialog extends ConnectedElement {
+  /**
+   * @type {Promise<Tabs>}
+   */
+  @queryAsync('sp-tabs')
+  accessor tabs;
+
+  /**
+   * @type {Promise<DialogBase>}
+   */
+  @queryAsync('sp-dialog-base')
+  accessor dialog;
+
+  /**
+   * The onboarding items
+   * @type {OnboardingItem[]}
+   */
+  @state()
+  accessor items;
+
+  /**
+   * The theme
+   * @type {'dark' | 'light'}
+   */
+  @state()
+  accessor theme;
+
+  /**
+   * The selected index
+   * @type {number}
+   */
+  @state()
+  accessor selectedIndex = 0;
+
+  static get styles() {
+    return [style];
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+
+    this.fetchIndex();
+    // Listen for theme changes
+    window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', this.onThemeChange);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    window.matchMedia('(prefers-color-scheme: dark)').removeEventListener('change', this.onThemeChange);
+  }
+
+  /**
+   * Handle theme changes
+   */
+  onThemeChange = () => {
+    /* istanbul ignore next */
+    this.requestUpdate();
+  };
+
+  /**
+   * Fetch the onboarding index
+   */
+  async fetchIndex() {
+    const resp = await fetch(`${TOOLS_ORIGIN}/sidekick/query-index.json`);
+    if (!resp.ok) {
+      this.remove();
+      return;
+    }
+
+    const index = await resp.json();
+    const onboardingItems = index.data
+      .filter(({ category }) => category === 'onboarding')
```

Comment:
We should add a filter for the current user language and only load the paths with that prefix. This will allow us to localize the onboarding content.